### PR TITLE
Benchmark for insert into `reported` table

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Benchmark for insert into `reported` table for two version of such table. Will be expanded later.

## Type of change

- Benchmarks (no changes in the code)

## Testing steps

Can be run locally against any database.

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by joselsegura — APPROVED** [review]

File: db_benchmark_test.go [hunk 1]
```diff
@@ -109,30 +187,156 @@ func setup(b *testing.B) *sql.DB {
...
+		}
 	}
 }
+
+// BenchmarkInsertIntoReportedTableV1 checks the speed of inserting into
+// reported table without event_type column
+func BenchmarkInsertIntoReportedTableV1(b *testing.B) {
+	report := ""
+
+	runBenchmarkInsertIntoReportedTableV1(b, 1, &report)
```
Comment (juandspy) [inline db_benchmark_test.go:333]: why is `scaleFactor` 1?

Comment (tisnik) [reply db_benchmark_test.go:333]: We need to run benchmark with tables of different sizes. So changing N via CLI flags won't be sufficient there

Comment (juandspy) [reply db_benchmark_test.go:333]: yes, but why 1 and not 10 f.e?

File: db_benchmark_test.go [hunk 2]
```diff
@@ -109,30 +187,156 @@ func setup(b *testing.B) *sql.DB {
...
+		}
 	}
 }
 
-// BenchmarkBar is dummy ATM
-func BenchmarkBar(b *testing.B) {
+// runBenchmarkInsertIntoReportedTableV2 function perform several inserts into
+// reported table v2. In case of any error detected, benchmarks fail
+// immediatelly.
+func runBenchmarkInsertIntoReportedTableV2(b *testing.B, scaleFactor int, report *string) {
```
Comment (epapbak) [inline db_benchmark_test.go:302]: what about having only on `runBenchmarkInsertIntoReportedTable` function and passing the `insertIntoReported` function as an argument too?

File: db_benchmark_test.go [hunk 3]
```diff
@@ -24,21 +24,94 @@ package main_test
...
+                    state             integer not null,
+                    report            varchar not null,
+                    updated_at        timestamp not null,
+                    notified_at       timestamp not null,
+		    error_log         varchar,
+                
+                    PRIMARY KEY (org_id, cluster, notified_at),
+                    CONSTRAINT fk_notification_type
+                        foreign key(notification_type)
+                        references notification_types(id),
```
Comment (juandspy) [inline db_benchmark_test.go:67]: It is necessary to create these tables too:
- notification_types
- states
isn't it? I can't run the benchmarks locally.

Comment (juandspy) [reply db_benchmark_test.go:67]: also, if you want to run this using compose, we have to change the admin password to `postgres` [here](https://github.com/RedHatInsights/ccx-notification-writer/blob/master/docker-compose.yml)

**Review by dpensi — APPROVED** [review]

File: db_benchmark_test.go (hunk 3 above)
Comment (tisnik) [reply db_benchmark_test.go:67]: it will run benchmarks, it's prepared to be run against migrated database

Comment (tisnik) [reply db_benchmark_test.go:67]: OTOH yes, we might probably start the migration from benchmarks, if you really wish to

File: db_benchmark_test.go (hunk 1 above)
Comment (tisnik) [reply db_benchmark_test.go:333]: ah sure, as I said, there will be much more benchmarks prepared later (today?)

File: db_benchmark_test.go (hunk 2 above)
Comment (tisnik) [reply db_benchmark_test.go:302]: good advice! will do it

File: db_benchmark_test.go (hunk 3 above)
Comment (juandspy) [reply db_benchmark_test.go:67]: hmm, if the migration is not included I'd add a brief explanation on what is needed for this benchmarks to run. Just a couple of lines in the code with the commands to run

File: db_benchmark_test.go (hunk 1 above)
Comment (juandspy) [reply db_benchmark_test.go:333]: nice!

File: db_benchmark_test.go (hunk 2 above)
Comment (tisnik) [reply db_benchmark_test.go]: implemented

**Review by epapbak — APPROVED** [review]
6 params is fine with me in this context =)

File: db_benchmark_test.go (hunk 3 above)
Comment (tisnik) [reply db_benchmark_test.go:73]: added

**Review by juandspy — APPROVED** [review]

**Pull request merged**